### PR TITLE
[RFC Don't merge yet] Fix concurrent map read/writes

### PR DESCRIPTION
**Pull request by tmarcu**
Signed-off-by: Tudor Marcu <tudor.marcu@intel.com>

### CODE REVIEW

File: builder/bundles.go [hunk 1]
```diff
@@ -240,6 +243,8 @@ func resolveFiles(numWorkers int, set bundleSet, bundleRepoPkgs map[string]map[s
 				errorCh <- err
 				break
 			}
+			pkgMutex.Unlock()
```
Comment (matthewrsj) [inline builder/bundles.go]: move this to before the error checking.

Comment (matthewrsj) [reply builder/bundles.go]: Actually, this doesn't need a mutex, it just needs a different name for the `err` variable so all goroutines aren't using the same one.

File: builder/bundles.go [hunk 2]
```diff
@@ -383,6 +389,7 @@ func resolvePackages(numWorkers int, set bundleSet, packagerCmd []string, emptyD
 					bundle.AllPackages[pkg] = true
 				}
 			}
+			pkgMutex2.Unlock()
```
Comment (matthewrsj) [inline builder/bundles.go]: Don't even need these.

File: builder/bundles.go [hunk 3]
```diff
@@ -230,16 +230,18 @@ func resolveFiles(numWorkers int, set bundleSet, bundleRepoPkgs map[string]map[s
...
+			e = resolveFilesForBundle(bundle, bundleRepoPkgs[bundle.Name], packagerCmd)
+			if e != nil {
 				// break on the first error we get
 				// causes wg.Done to be called and the worker to exit
 				// the break is important so we don't overflow the errorCh
-				errorCh <- err
+				errorCh <- e
 				break
 			}
+
```
Comment (matthewrsj) [inline builder/bundles.go]: extra whitespace?

Comment (tmarcu) [reply builder/bundles.go]: yeah I formatted with go fmt just now and repushed :) thx

File: builder/bundles.go [hunk 4]
```diff
@@ -20,6 +20,8 @@ import (
 	"github.com/pkg/errors"
 )
 
+var pkgMutex sync.RWMutex
```
Comment (matthewrsj) [inline builder/bundles.go]: Can you move this declaration closer to the function that uses it?

Comment (tmarcu) [reply builder/bundles.go]: yes

**Pull request closed without merging**